### ISSUE TITLE
Add metadata tags for each pillar

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1772]
This update adds a new tag `Azure.WAF/pillar` to each rule which is set to the WAF pillar.

Why are we doing this:

- This will allow customers to create WAF pillar specific baselines.
- PSRule for Azure intends to ship a managed quarterly baseline containing only security rules. #1633
- Improve the ability to provide PSRule for Azure recommendations in other languages.

To make this happen we will be updating all rules with this tag starting with the `Security` pillar.

Longer term we expect to remove the existing annotation for `pillar`. However in the short term both tag and annotation will exist.

Syntax:

```text
Azure.WAF/pillar: 'Security'
```

For example, in YAML:

```yaml
---
# Synopsis: Internet accessible Application Gateways should use protect endpoints with WAF.
apiVersion: github.com/microsoft/PSRule/v1
kind: Rule
metadata:
  name: Azure.AppGw.UseWAF
  ref: AZR-000063
  tags:
    release: 'GA'
    ruleSet: '2020_06'
    Azure.WAF/pillar: 'Security'
  labels:
    Azure.ASB.v3/control: 'NS-6'
spec:
  with:
  - Azure.IsAppGwPublic
  condition:
    field: properties.sku.tier
    in:
    - WAF
    - WAF_v2
```

For example in PowerShell:

```powershell
# Synopsis: App Service should reject TLS versions older then 1.2.
Rule 'Azure.AppService.MinTLS' -Ref 'AZR-000073' -Type 'Microsoft.Web/sites', 'Microsoft.Web/sites/slots' -Tag @{ release = 'GA'; ruleSet = '2020_06'; 'Azure.WAF/pillar' = 'Security'; } -Labels @{ 'Azure.ASB.v3/control' = 'DP-3' } {
    $siteConfigs = @(GetWebSiteConfig);
    if ($siteConfigs.Length -eq 0) {
        return $Assert.
            HasFieldValue($TargetObject, 'properties.siteConfig.minTlsVersion', '1.2').
            ReasonFrom('properties.siteConfig.minTlsVersion', $LocalizedData.MinTLSVersion, $TargetObject.properties.siteConfig.minTlsVersion);
    }
    foreach ($siteConfig in $siteConfigs) {
        $path = $siteConfig._PSRule.path;
        $Assert.
            HasFieldValue($siteConfig, 'properties.minTlsVersion', '1.2').
            ReasonFrom('properties.minTlsVersion', $LocalizedData.MinTLSVersion, $siteConfig.properties.minTlsVersion).PathPrefix($path);
    }
}
```

Some of this work is already complete related to #1610
